src/plugins/intel_npu/src/plugin/npuw/just_sync_infer_request.cpp

```diff
@@ -377,6 +379,25 @@ void ov::npuw::JustInferRequest::set_tensor(const ov::Output<const ov::Node>& po
             }
         }
     }
+
+    // Check if setting input tensor
+    for (std::size_t i = 0; i < m_npuw_model->inputs().size(); ++i) {
+        if (m_npuw_model->inputs()[i] == port) {
```

Comment:
our set_tensor has O(n) it is pity :)

Reply:
Agree, but it was already O(n) due to outputs processing